src/main/java/org/opensearch/security/identity/SecurityIndices.java

```diff
@@ -0,0 +1,166 @@
+package org.opensearch.security.identity;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.EnumMap;
+
+import com.google.common.io.Resources;
+import org.apache.commons.codec.Charsets;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+
+public class SecurityIndices {
+    protected Logger logger = LogManager.getLogger(getClass());
+
+    public final static String SCHEDULED_JOB_IDENTITY_INDEX = ".opendistro-security-scheduled-job-identity";
+    public static final String SCHEDULED_JOB_IDENTITY_INDEX_MAPPING_FILE = "mappings/scheduled-job-identity.json";
+    static final String META = "_meta";
+    private static final String SCHEMA_VERSION = "schema_version";
+    public static Integer NO_SCHEMA_VERSION = 0;
+    // minimum shards of the scheduled job identity index
+    public static int minJobIndexReplicas = 1;
+    // maximum shards of the scheduled job identity index
+    public static int maxJobIndexReplicas = 20;
+    private final Client client;
+    private final ClusterService cs;
+    // keep track of whether the mapping version is up-to-date
+    private EnumMap<SecurityIndex, IndexState> indexStates;
+
+    class IndexState {
```

Comment:
Could you please explain the purpose of this class?

Reply:
This is based off some similar code in AD. I will take a deeper dive and reply back with findings.